### PR TITLE
Added HEAD queries + thread_id support

### PR DESCRIPTION
**Pull request by dingus9**
Left some testing to be done in upcoming artman
HEAD supports similar API to get but just returns counts
Removed /gman from all gman routes to no /thread/... or /task/... etc.
Increased test coverage of gman.py to 99% will allow the remaining coverage to be covered in artman branch as I've tested those routes and features there already.

### CODE REVIEW

File: piedpiper_gman/gman.py [hunk 1]
```diff
@@ -10,108 +12,287 @@
...
+                         .order_by(TaskEvent.timestamp)]
+
+    def head(self, thread_id=None, events=None, task_id=None):
+
+        if thread_id:
+            task_events = self.get_event_thread(thread_id)
+        elif task_id:
+            task_events = self.get_task_events(task_id)
+
+        if len(task_events):
```
Comment (ragingpastry) [inline piedpiper_gman/gman.py:175]: Should we check if task_events exists before checking its length? Or maybe set it to empty first. Currently gman will return a 500 using:
curl --head localhost/gman

Comment (dingus9) [reply piedpiper_gman/gman.py:175]: yes, yes we should... fixing

File: piedpiper_gman/gman.py [hunk 2]
```diff
@@ -10,108 +12,287 @@
...
+            try:
+                self._event.data.timestamp = datetime.datetime.now()
+            except AttributeError:
+                self._event.data['timestamp'] = datetime.datetime.now()
+
+            if self._event.errors:
+                self.errors.extend(self._event.errors)
+
+            if self.raw_data.get('status', '') in ('started', 'received'):
+                self.errors.add('status', 'Updates may not be value \'started\'')
```
Comment (ragingpastry) [inline piedpiper_gman/gman.py]: Should this say "Updates may not be value \'started\' or \'received\'' ?

File: piedpiper_gman/gman.py [hunk 3]
```diff
@@ -10,108 +12,305 @@
...
+                self._event.data.timestamp = datetime.datetime.now()
+            except AttributeError:
+                self._event.data['timestamp'] = datetime.datetime.now()
+
+            if self._event.errors:
+                self.errors.extend(self._event.errors)
+
+            if self.raw_data.get('status', '') in ('started', 'received'):
+                self.errors.add('status',
+                                'Updates may not be value \'started\' or \'started\'')
```
Comment (ragingpastry) [inline piedpiper_gman/gman.py]: Started or received ?

File: piedpiper_gman/gman.py (hunk 1 above)
Comment (dingus9) [reply piedpiper_gman/gman.py:175]: fixed although it's not updating showing that I did... IDK

**Comment by dingus9** [comment]
okay I think I addressed those issues

**Review by ragingpastry — APPROVED** [review]

**Pull request merged**